week3/project/ecommerce/src/pages/ProductsPage/products/product.jsx

```diff
@@ -0,0 +1,39 @@
+// import { useContext, useState } from 'react'
+import { Link } from 'react-router-dom';
+// import svgReg from '../../../images/heart-regular.svg';
+// import svgSol from '../../../images/heart-solid.svg';
+// import { FavoriteIdItemContext } from '../../../contexts/FavoriteIdItemContext';
+import useFavoriteItems from '../../../hooks/useFavoriteItems';
+
+export default function Product({ item }) {
+
+   const { svgIcon, favoriteIdHandler } = useFavoriteItems(item.id)
+
+   // const { idFavoriteItem, setIdFavoriteItem } = useContext(FavoriteIdItemContext);
+   // const [svgIcon, setSvgIcon] = useState(idFavoriteItem.includes(item.id) ? svgSol : svgReg)
+
+
+   // function favoriteIdHandler(itemId) {
+   //    if (!idFavoriteItem.includes(itemId)) {
+   //       setIdFavoriteItem((prev) => [...prev, itemId])
+   //       setSvgIcon(svgSol)
+   //    }
+   //    else {
+   //       setIdFavoriteItem((prev) => prev.filter(id => id !== itemId))
+   //       setSvgIcon(svgReg)
+   //    }
+   // }
```

Comment:
brilliant
and this is a great example to show how custom hooks can save you time and make you code cleaner and easier to read and understand even if you open it later after five months for example